### PR TITLE
set filename if azure pipe is used as port

### LINKED ISSUE #9: serverPort is not compatible to Azure App Service pipes
Hi,

have a problem running a node.js app (OpenTok RTC V2) on Azure App Service.

```
const serverPort = process.env.PORT || 8123;
const SwaggerServer = require('swagger-boilerplate').Server;
const server = new SwaggerServer({
  apiFile: './api.yml',
  modulePath: `${__dirname}/server/`,
  appName: 'OpenTokRTC Main',
  serverPort: serverPort,
  staticOptions: {
    dotfiles: 'ignore',
    extensions: ['jpg'],
    index: false,
    redirect: false,
  },
});

server.start();
```

Problem is that serverPort is something like "\\.\pipe\25387797-4dbd-4475-91d2-cf8873ac3aad" in Azure App Service, cause there is no integer port used but a pipe that the entry port gets forwarded to.

So I will get error: "ENOENT: no such file or directory, open 'C:\home\site\wwwroot\otrtc_\pipe\e8dbf4f3-3786-4c4a-8cc3-a460c2926edb.pid'" because the filename for the .pid file is malformed as a path that does not exist.

Unfortunatly azure app service is not accepting requests if I change port to some other. Can not get node.js to run on custom port.

Is there allready a solution for that problem? 

Thanks in advance,
Sven

### PR DESCRIPTION
**Pull request by DarthSonic**
Fixes #9 

Did not find any more references to the *.pid file.

### CODE REVIEW

**Comment by DarthSonic** [comment]
This is my first ever fork and pull-request. Please apologize if I have done anything wrong with the pull request.

**Comment by AntonioMA** [comment]
Sorry for the very late reply. I hadn't enabled the notifications for this repository, and only saw the PR by pure chance.

File: index.js [hunk 1]
```diff
@@ -6,13 +6,13 @@ var JSONPersistenceProvider = require('./lib/jsonPersistenceProvider');
...
+//var RestAPI = require('./lib/restAPI');
 module.exports = {
   Server,
   JSONPersistenceProvider,
   ServerPersistence,
   Utils,
   ErrorInfo,
   App,
-  RestAPI,
+  //RestAPI,
```
Comment (AntonioMA) [inline index.js]: I've added the file. For some reason (this was 3 years ago 😄 ) I had the file locally, and pushed it to npm, but forgot to add and push it here. So you can reenable this.

File: lib/server.js [hunk 2]
```diff
@@ -102,9 +102,16 @@ module.exports = function(aOptions) {
       serverType.createServer.apply(serverType, serverParams).
         listen(serverPort);
 
+      // Azure App Service ports are actually pipes like "\\.\pipe\3212d91e-bc1a-4720-8792-49cf5f9eba86"
+      let serverPortRaw = String.raw`${options.serverPort}`;
+      let pidStreamId = serverPortRaw.replace("\\\\.\\pipe\\", '');
+
+      //logger.log('serverPort is ', serverPortRaw);
```
Comment (AntonioMA) [inline lib/server.js]: Please remove dead code (if you add lines for debug or whatever, just remove them afterwards, don't comment out them).

File: lib/server.js [hunk 3]
```diff
@@ -102,9 +102,16 @@ module.exports = function(aOptions) {
       serverType.createServer.apply(serverType, serverParams).
         listen(serverPort);
 
+      // Azure App Service ports are actually pipes like "\\.\pipe\3212d91e-bc1a-4720-8792-49cf5f9eba86"
+      let serverPortRaw = String.raw`${options.serverPort}`;
```
Comment (AntonioMA) [inline lib/server.js]: Use `const` here and on the following line instead of `let`.

**Review by AntonioMA — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a couple of small changes and I can land this. Also if you want me to publish this as a new version on NPM, please update the version number on package.json (to 0.1.7) also.

File: index.js (hunk 1 above)
Comment (DarthSonic) [reply index.js]: added RestAPI again

File: lib/server.js (hunk 2 above)
Comment (DarthSonic) [reply lib/server.js]: removed dead code

File: lib/server.js (hunk 3 above)
Comment (DarthSonic) [reply lib/server.js]: used const instead of let

**Comment by DarthSonic** [comment]
Please review. I did the requested changes.

**Review by AntonioMA — APPROVED** [review]

**Comment by AntonioMA** [comment]
> Please review. I did the requested changes.

Done and merged. Thank you!

**Pull request merged**